src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,76 @@
-import React from 'react';
+import { useEffect, useState } from 'react';
+
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { useAppSelector } from '../../hooks/useAppSelector';
+import { useDispatch } from 'react-redux';
+import { currentTodoSlice } from '../../features/currentTodo';
+
+export const TodoModal = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [user, setUser] = useState<User | null>(null);
+  const dispatch = useDispatch();
+  const selectedTodo = useAppSelector(state => state.currentTodo);
+
+  useEffect(() => {
+    if (selectedTodo) {
+      setIsLoading(true);
+      getUser(selectedTodo?.userId)
```

Comment:
The optional chaining `?.` is unnecessary here since `selectedTodo` is already checked for truthiness in the `if` statement. This could lead to confusion or unnecessary API calls if `selectedTodo` is null or undefined.